presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -195,7 +198,16 @@ private HiveUtil()
         jobConf.set("io.compression.codecs", codecs.stream().collect(joining(",")));
 
         try {
-            return inputFormat.getRecordReader(fileSplit, jobConf, Reporter.NULL);
+            RecordReader<?, ?> recordReader = inputFormat.getRecordReader(fileSplit, jobConf, Reporter.NULL);
+
+            int headerCount = getHeaderCount(schema);
+            if (headerCount > 0) {
+                WritableComparable key = (WritableComparable) recordReader.createKey();
+                Writable value = (Writable) recordReader.createValue();
+                Utilities.skipHeader((RecordReader<WritableComparable, Writable>) recordReader, headerCount, key, value);
```

Comment:
move the `RecordReader<WritableComparable, Writable>` cast to `recordReader` declaration. (should this actually be `RecordReader<? extends WritableComparable, ? extends Writable>`? Can this be anything else?)